cares-center-inventory-Song/crud.py

```diff
@@ -0,0 +1,242 @@
+
+def get_items_by_donor_name(db: Session, donor_name: str):
+    return db.query(models.Item).join(models.Donor).filter(models.Donor.donor_name == donor_name).all()
+
+def get_items_by_point_interval(db: Session, lhs: int, rhs: int):
+    return db.query(models.Item).filter(and_(models.Item.item_points >= lhs, models.Item.item_points <= rhs)).all()
+
+def update_item_name_by_item_id(db: Session, item_id: str, new_item_name: str):
+    db.query(models.Item).filter(models.Item.item_id == item_id).update({models.Item.item_name: new_item_name})
+    db.commit()
+
+def update_item_front_quantity_by_item_id(db: Session, item_id: str, new_item_front_quantity):
+    db.query(models.Item).filter(models.Item.item_id == item_id).update({models.Item.item_front_quantity: new_item_front_quantity})
+    db.commit()
+
+def delete_item_by_item_id(db: Session, item_id: str):
+    item_to_delete = db.query(models.Item).filter(models.Item.item_id == item_id).first()
+    db.delete(item_to_delete)
+    db.commit()
+
+# replenish_history
+def create_replenish_history(db: Session, replenish_history: schemas.ReplenishHistoryCreate):
+    db_replenish_history = models.ReplenishHistory(
+        replenish_date = replenish_history.replenish_date,
+        manager_id = replenish_history.manager_id,
+    )
+    db.add(db_replenish_history)
+    db.commit()
+    db.refresh(db_replenish_history)
+    return db_replenish_history
+
+def get_replenish_history_by_replenish_id(db: Session, replenish_id: str):
+    return db.query(models.ReplenishHistory).filter(models.ReplenishHistory.replenish_id == replenish_id).first()
+
+def get_replenish_histories_by_replenish_date(db: Session, replenish_date: str):
+    return db.query(models.ReplenishHistory).filter(models.ReplenishHistory.replenish_date == replenish_date).all()
+
+def get_replenish_histories_by_manager_id(db: Session, manager_id: str):
+    return db.query(models.ReplenishHistory).filter(models.ReplenishHistory.manager_id == manager_id).all()
+
+def delete_replenish_history_by_replenish_id(db: Session, replenish_id: str):
+    replenish_history_to_delete = db.query(models.ReplenishHistory).filter(models.ReplenishHistory.replenish_id == replenish_id).first()
+    db.delete(replenish_history_to_delete)
+    db.commit()
+
+# replenish_item
+def create_replenish_item(db: Session, replenish_item = schemas.ReplenishItemCreate):
+    db_replenish_item = models.ReplenishItem(
+        item_id = replenish_item.item_id,
+        replenish_id = replenish_item.replenish_id,
+        replenish_quantity = replenish_item.replenish_quantity
+    )
+    db.add(db_replenish_item)
+    db.commit()
+    db.refresh(db_replenish_item)
+
+def get_replenish_items_by_replenish_id(db: Session, replenish_id: str):
+    return db.query(models.ReplenishItem).filter(models.ReplenishItem.replenish_id == replenish_id).all()
+
+def delete_replenish_items_by_item_id(db: Session, item_id: str):
```

Comment:
Combine this function with the one starting on line 152, we want to take an item_id and replenish_id so we can delete a specific items from a specific replenishment